homeassistant/components/fing/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow file."""
+
+import logging
+from typing import Any
+
+from fing_agent_api import FingAgent
+import httpx
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.core import HomeAssistant
+
+from .const import AGENT_IP, AGENT_KEY, AGENT_PORT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_data_schema(
+    hass: HomeAssistant, user_input: dict[str, Any] | None = None
+) -> vol.Schema:
+    """Get a schema with default values."""
+
+    if user_input is None:
+        return vol.Schema(
+            {
+                vol.Required(AGENT_IP): str,
+                vol.Required(AGENT_PORT, default="49090"): str,
+                vol.Required(AGENT_KEY): str,
+            }
+        )
+
+    return vol.Schema(
+        {
+            vol.Required(AGENT_IP, default=user_input.get(AGENT_IP)): str,
+            vol.Required(AGENT_PORT, default=user_input.get(AGENT_PORT)): str,
+            vol.Required(AGENT_KEY, default=user_input.get(AGENT_KEY)): str,
+        }
+    )
+
+
+class FingConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Fing config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Set up user step."""
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+
+        if user_input is not None:
+            try:
+                existing_entries = [
+                    entry
+                    for entry in self._async_current_entries()
+                    if entry.data.get(AGENT_IP) == user_input[AGENT_IP]
+                ]
+                if existing_entries:
+                    return self.async_abort(reason="already_configured")
```

Comment:
This can be done with `self._abort_entries_match({AGENT_IP: user_input[AGENT_IP]})`

Let's move it out of the try block